server/src/Api/Handlers.hs

```diff
@@ -75,13 +80,59 @@ import Types (
   hashLedgerScript,
  )
 
+--------------------------------------------------------------------------------
+-- Handlers
+--------------------------------------------------------------------------------
+
 estimateTxFees :: WitnessCount -> Cbor -> AppM Fee
 estimateTxFees (WitnessCount numWits) cbor = do
-  decoded <-
-    either (throwM . CborDecode) pure $
-      decodeCborTx cbor
-  pparams <- asks protocolParams
-  pure . Fee $ estimateFee pparams numWits decoded
+  tx <- decodeCborTx cbor & either (throwM . CborDecode) pure
+  case tx of
```

Comment:
Why not just use a pattern directly in the binding? `C.Tx txBody' keyWits <- decode...`

Reply:
Hmmm apparently that's a fallible pattern, interesting